examples/fluids/src/dirichlet.c

```diff
@@ -0,0 +1,194 @@
+                                     num_elem * elem_size * num_comp_x,
+                                     CEED_STRIDES_BACKEND, &elem_restr_x_stored);
+    CeedElemRestrictionCreateVector(elem_restr_x_stored, &x_stored, NULL);
+
+    CeedElemRestrictionCreateStrided(ceed, num_elem, elem_size, 1,
+                                     num_elem * elem_size,
+                                     CEED_STRIDES_BACKEND, &elem_restr_scale);
+    CeedElemRestrictionCreateVector(elem_restr_scale, &scale_stored, NULL);
+
+    CeedVectorCreate(ceed, q_data_size_sur*num_elem*elem_size, &q_data_sur);
+
+    // -- Setup Operator
+    CeedOperatorCreate(ceed, qf_setup, NULL, NULL, &op_setup);
+    CeedOperatorSetName(op_setup, "surface geometric data");
+    CeedOperatorSetField(op_setup, "x", elem_restr_x_sur, basis_x_to_q_sur,
+                         CEED_VECTOR_ACTIVE);
+    CeedOperatorSetField(op_setup, "multiplicity", elem_restr_q_sur,
+                         CEED_BASIS_COLLOCATED, multiplicity);
+    CeedOperatorSetField(op_setup, "x stored", elem_restr_x_stored,
+                         CEED_BASIS_COLLOCATED, x_stored);
+    CeedOperatorSetField(op_setup, "scale", elem_restr_scale, CEED_BASIS_COLLOCATED,
+                         scale_stored);
+
+    // -- Compute geometric factors
+    CeedOperatorApply(op_setup, ceed_data->x_coord, CEED_VECTOR_NONE,
+                      CEED_REQUEST_IMMEDIATE);
+
+    // -- Compute QData for the surface
+    CeedOperatorCreate(ceed, ceed_data->qf_setup_sur, NULL, NULL, &op_setup_sur);
+    CeedOperatorSetField(op_setup_sur, "dx", elem_restr_x_sur,
+                         ceed_data->basis_x_sur, CEED_VECTOR_ACTIVE);
+    CeedOperatorSetField(op_setup_sur, "weight", CEED_ELEMRESTRICTION_NONE,
+                         ceed_data->basis_x_sur, CEED_VECTOR_NONE);
+    CeedOperatorSetField(op_setup_sur, "surface qdata", elem_restr_qd_sur,
+                         CEED_BASIS_COLLOCATED, CEED_VECTOR_ACTIVE);
+
+    CeedOperatorApply(op_setup_sur, ceed_data->x_coord, q_data_sur,
+                      CEED_REQUEST_IMMEDIATE);
+
+    // -- Setup BC QFunctions
+    SetupStrongSTG_QF(ceed, problem, num_comp_x, num_comp_q, q_data_size_sur,
+                      &qf_strongbc);
+    CeedOperatorCreate(ceed, qf_strongbc, NULL, NULL, &op_dirichlet_sub);
+    CeedOperatorSetName(op_dirichlet_sub, "Strong STG");
+
+    CeedOperatorSetField(op_dirichlet_sub, "surface qdata", elem_restr_qd_sur,
+                         CEED_BASIS_COLLOCATED, q_data_sur);
+    CeedOperatorSetField(op_dirichlet_sub, "x", elem_restr_x_stored,
+                         CEED_BASIS_COLLOCATED, x_stored);
+    CeedOperatorSetField(op_dirichlet_sub, "scale", elem_restr_scale,
+                         CEED_BASIS_COLLOCATED, scale_stored);
+    CeedOperatorSetField(op_dirichlet_sub, "q", elem_restr_q_sur,
+                         CEED_BASIS_COLLOCATED, CEED_VECTOR_ACTIVE);
+    CeedOperatorSetNumQuadraturePoints(op_dirichlet_sub, elem_size);
```

Comment:
The number of quadratre points seems correct hore

Reply:
I think this level must all be correct because the cuda/shared (and CPU) backends have correct results and no extra quadrature points.